Add task1

mpozdeev opened the pull request.

mpozdeev committed:
Add task1 Utilities class test

Added methods: concatenateWords() and computeFactorial() and their unit tests

mpozdeev committed:
Changed the parent pom

Now this is a multi-module project.
Add dependency junit-4.12

mpozdeev committed:
Add task3 Mocking

Added mock objects, behavior for mock, and tests

IliaIsakhin commented on unit-testing/src/main/java/app/Utils.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+
+    private static boolean isWordExist(String word) {
```
return word != null;

mpozdeev replied:
Ого. Совсем коротко. Тогда смысла существования этого метода в принципе нет))) Он и так использовался единожды, а с такой записью только лишний код.
![1](https://user-images.githubusercontent.com/40239712/67008418-78c24a80-f0fa-11e9-8571-e9d92d6c857a.PNG)
![2](https://user-images.githubusercontent.com/40239712/67008464-9394bf00-f0fa-11e9-8606-12ae8c92f418.PNG)

IliaIsakhin commented on unit-testing/src/main/java/app/Utils.java:
```diff
@@ -0,0 +1,57 @@
+        }
+    }
+
+    static long computeFactorial(int number) {
```
Очень усложнено, упрости эту функцию

mpozdeev replied:
Упростил

IliaIsakhin commented on unit-testing/src/main/java/app/Utils.java:
```diff
@@ -0,0 +1,57 @@
+                throw new IllegalArgumentException("Задано отрицательное число");
+            }
+        } catch (IllegalArgumentException e){
+            result = -1;
```
В итоге факториал от -1 будет -1?

mpozdeev replied:
Вот-же... Исправил

IliaIsakhin commented on unit-testing/src/test/java/app/UtilsTest.java:
```diff
@@ -0,0 +1,185 @@
+     */
+
+    @Test
+    public void isBothWordsExist() {
```
Очень много повторяющихся тестов, вынести в параметризованный

mpozdeev committed:
Merge branch 'master' into dev/task1-classTest

mpozdeev committed:
Test changes

Added parameterized test for concatenateWords() method.
Method testing is divided into two separate test files.

mpozdeev committed:
Merge branch 'dev/task3-mocking' into dev/task1-classTest
